core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/components/dot-binary-field-editor/dot-binary-field-editor.component.ts

```diff
@@ -181,11 +181,36 @@ export class DotBinaryFieldEditorComponent implements OnInit, OnChanges {
     }
 
     private setEditorLanguage(fileName = '') {
-        const fileExtension = fileName?.includes('.') ? fileName.split('.').pop() : '';
+        const fileExtension = this.extractFileExtension(fileName);
+
+        if (fileExtension === 'vtl') {
+            this.setVelocityLanguage();
+        } else {
+            this.updateLanguageForFileExtension(fileExtension);
+        }
+
+        this.validateFileType(fileExtension);
+        this.cd.detectChanges();
+    }
+
+    private extractFileExtension(fileName: string) {
+        return fileName?.includes('.') ? fileName.split('.').pop() : '';
+    }
+
+    private setVelocityLanguage() {
+        this.mimeType = 'text/x-velocity';
+        this.extension = 'vtl';
+        this.updateEditorLanguage('html'); //Force html highlighting for .vtl files
+    }
+
+    private updateLanguageForFileExtension(fileExtension: string) {
         const { id, mimetypes, extensions } = this.getLanguage(fileExtension) || {};
         this.mimeType = mimetypes?.[0];
```

Comment:
@valentinogiardino Should the mimeType have a default value if `mimetypes?.[0]` is undefined?